main.nf

```diff
@@ -1,234 +1,292 @@
+    SNP & SEQ Run folder QC pipeline.
+
+    This workflow runs the following tools on a run folder:
+        * InterOp summary    (http://illumina.github.io/interop/example_summary.html)
+        * FastQC             (https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
+        * FastqScreen        (https://www.bioinformatics.babraham.ac.uk/projects/fastq_screen/)
+        * CheckQC            (https://github.com/Molmed/checkQC)
+        * MultiQC            (https://multiqc.info/)
+
+    Usage:
+        # Using parameters supplied in a config
+        nextflow run -c custom.config -profile snpseq main.nf
+
+        # Using parameters supplied on the command line
+        nextflow run -profile snpseq main.nf \\
+            --run_folder '/path/to/runfolder' \\
+            --fastqscreen_databases '/path/to/databases' \\
+            --checkqc_config '/path/to/checkqc.config'
+
+    Mandatory parameters:
+        --run_folder                        The folder to quality check
+
+    Optional parameters:
+        --result_dir                        Path to write results (default: results)
+        --bcl2fastq_outdir                  Folder name to check for fastq.gz files and demultiplexing stats (default: Unaligned)
+        --checkqc_config                    Configuration file for CheckQC
+        --assets_dir                        Location of project assests (default: "\$baseDir/assets").
+        --config_dir                        Location of project configuration files (default: "\$baseDir/config").
+        --script_dir                        Location of project scripts (default: "\$baseDir/bin")
+
+        --help                              Print this help message.
+
+    Notes:
+        * Always quote paths that are parameters to nextflow e.g. '/path/to/file'
+
+    """
+}
+
+if (params.help || !params.run_folder){
+    helpMessage()
+    exit 0
+}
+
+workflow {
+
+    main:
+    get_run_folder(params.run_folder) | check_run_quality
+
+    publish:
+    check_run_quality.out.projectqc            to: "${params.result_dir}/multiqc_by_project"
+    check_run_quality.out.flowcellqc           to: "${params.result_dir}/multiqc_by_flowcell"
+
+}
+
+workflow.onComplete {
+	log.info ( workflow.success ? "\nDone! Open the reports in your browser.\n" : "Oops .. something went wrong." )
+}
+
+def get_run_folder(run_folder) {
+
+    Channel.value(file(run_folder))
+        .ifEmpty { "Error: No run folder (--run_folder) given."; exit 1 }
+
+}
+
+def get_project_and_reads(run_folder) {
+
+    Channel
+        .fromPath("${run_folder}/${params.bcl2fastq_outdir}/**.fastq.gz", maxDepth: 10 )
+        .filter( ~/.*_[^I]\d_001\.fastq\.gz$/ )
+        .ifEmpty { "Error: No fastq files found under ${run_folder}/ !\n"; exit 1 }
+        .map {
+            it.toString().indexOf('Undetermined') > 0 ?
+                ['NoProject', it] :
+                [(it.toString() =~ /^.*\/Unaligned\/([^\/]+)\/.*\.fastq\.gz$/)[0][1],it]
```

Comment:
Unaligned should be `bcl2fastq_outdir`.

Reply:
Good catch. Changed "Unaligned" to the variable.